kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,220 @@
+    "warp_image_tps"
+]
+
+# utilities for computing thin plate spline transforms
+
+
+def _pair_square_euclidean(tensor1: torch.Tensor, tensor2: torch.Tensor) -> torch.Tensor:
+    r"""Compute the pairwise squared euclidean distance matrices :math:`(B, N, M)` between two tensors
+    with shapes (B, N, C) and (B, M, C)."""
+    # ||t1-t2||^2 = (t1-t2)^T(t1-t2) = t1^T*t1 + t2^T*t2 - 2*t1^T*t2
+    t1_sq: torch.Tensor = tensor1.mul(tensor1).sum(dim=-1, keepdim=True)
+    t2_sq: torch.Tensor = tensor2.mul(tensor2).sum(dim=-1, keepdim=True).transpose(1, 2)
+    t1_t2: torch.Tensor = tensor1.matmul(tensor2.transpose(1, 2))
+    square_dist: torch.Tensor = -2 * t1_t2 + t1_sq + t2_sq
+    square_dist: torch.Tensor = square_dist.clamp(min=0)  # handle possible numerical errors
+    return square_dist
+
+
+def _kernel_distance(squared_distances: torch.Tensor, eps: float = 1e-8) -> torch.Tensor:
+    r"""Compute the TPS kernel distance function: :math:`r^2 log(r)`, where `r` is the euclidean distance.
+    Since :math:`\log(r) = 1/2 \log(r^2)`, this function takes the squared distance matrix and calculates
+    :math:`0.5 r^2 log(r^2)`."""
+    # r^2 * log(r) = 1/2 * r^2 * log(r^2)
+    return 0.5 * squared_distances * squared_distances.add(eps).log()
+
+
+def get_tps_transform(points_src: torch.Tensor, points_dst: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+    r"""Compute the TPS transform parameters that warp source points to target points.
+
+    The input to this function is a tensor of :math:`(x, y)` source points :math:`(B, N, 2)` and a corresponding
+    tensor of target :math:`(x, y)` points :math:`(B, N, 2)`.
+
+    Args:
+        points_src (torch.Tensor): batch of source points :math:`(B, N, 2)` as :math:`(x, y)` coordinate vectors.
+        points_dst (torch.Tensor): batch of target points :math:`(B, N, 2)` as :math:`(x, y)` coordinate vectors.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]: :math:`(B, N, 2)` tensor of kernel weights and :math:`(B, 3, 2)`
+            tensor of affine weights. The last dimension contains the x-transform and y-transform weights
+            as seperate columns.
+
+    Example:
+        >>> points_src = torch.rand(1, 5, 2)
+        >>> points_dst = torch.rand(1, 5, 2)
+        >>> kernel_weights, affine_weights = get_tps_transform(points_src, points_dst)
+    """
+    if not isinstance(points_src, torch.Tensor):
+        raise TypeError(f"Input points_src is not torch.Tensor. Got {type(points_src)}")
+
+    if not isinstance(points_dst, torch.Tensor):
+        raise TypeError(f"Input points_dst is not torch.Tensor. Got {type(points_dst)}")
+
+    if not len(points_src.shape) == 3:
+        raise ValueError(f"Invalid shape for points_src, expected BxNx2. Got {points_src.shape}")
+
+    if not len(points_dst.shape) == 3:
+        raise ValueError(f"Invalid shape for points_dst, expected BxNx2. Got {points_dst.shape}")
+
+    device, dtype = _extract_device_dtype([points_src, points_dst])
```

Comment:
this function seems to break jit. Test in local to make sure that it works. What I would suggest is to get rid of this function and get directly the device/dtype from src or dst. In case they don't match, pytorch itself will raise an error that usually is easy for the user to understand what's going on.